### PR TITLE
updates and fixes

### PR DESCRIPTION
**Pull request by sgesin**
Mostly updates from 228/patch and bumping up the api versions.

With recent changes to sfdx we also needed to restructure the project to remove circular dependencies between force-app and examples directories. Now force-app contains only source code and labels while examples contains all the flexipages, classes, etc that are needed to make the Base Components examples app work. These are deployed sequentially based on the order in the sfdx-project.json file.

### CODE REVIEW

File: config/project-scratch-def.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "orgName": "Base Components Recipes",
     "edition": "Developer",
-    "hasSampleData": false,
+    "hasSampleData": true,
```
Comment (sgesin) [inline config/project-scratch-def.json:4]: Sample data is needed for some examples to work.

File: force-app/main/default/lwc/carouselImage/__tests__/carouselImage.spec.js [hunk 2]
```diff
@@ -7,9 +7,9 @@
 
 import { createElement } from 'lwc';
 import Element from 'c/carouselImage';
-import { axe, toHaveNoViolations } from 'jest-axe';
+import { registerSa11yMatcher } from '@sa11y/jest';
```
Comment (vbhargavmbis) [inline force-app/main/default/lwc/carouselImage/__tests__/carouselImage.spec.js:10]: @sgesin Does this (sa11y) actually work in this repo?

File: force-app/main/default/lwc/pillContainer/barePillContainer.html [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                </button></span
+            >
+        </template>
+        <template
+            for:each={pillViewModels}
+            for:item="pillViewModel"
+            for:index="index"
+        >
+            <!--
+             W-8012126: Temporarily adding slds-truncate so pills truncate on small form factors.
```
Comment (vbhargavmbis) [inline force-app/main/default/lwc/pillContainer/barePillContainer.html]: Is it ok to keep this?
I suppose it should be fine - we have more work item id's in the js files.

File: force-app/main/default/lwc/routingService/routingService.js [hunk 4]
```diff
@@ -56,8 +56,8 @@ export function updateRawLinkInfo(element, { url, target }) {
         // eslint-disable-next-line no-console
         console.error('url must be specified');
     }
-    if (target === '_blank') {
-        return new Promise(resolve => {
+    if (target && target !== '_self') {
+        return new Promise((resolve) => {
```
Comment (vbhargavmbis) [inline force-app/main/default/lwc/routingService/routingService.js:60]: I'm seeing a discrepancy in this file compared to the original file - https://github.com/salesforce/lightning-components/blob/master/ui-lightning-components/src/main/modules/lightning/routingService/routingService.js

@sgesin Can you confirm if the changes here are expected or if we need to update this file?

**Comment by vbhargavmbis** [comment]
@sgesin Just checking again - are the files under `optional/translations/` and `iconSvgTemplates/buildTemplates/` meant to be part of this repo?

File: force-app/main/default/lwc/routingService/routingService.js (hunk 4 above)
Comment (sgesin) [reply force-app/main/default/lwc/routingService/routingService.js:60]: @vbhargavmbis It should match with the 228/patch version:
https://github.com/salesforce/lightning-components/blob/228/patch/ui-lightning-components/src/main/modules/lightning/routingService/routingService.js

It seems like the only diffs between those two is formatting and comments getting stripped.

File: force-app/main/default/lwc/pillContainer/barePillContainer.html (hunk 3 above)
Comment (sgesin) [reply force-app/main/default/lwc/pillContainer/barePillContainer.html]: Good catch, I'll add a work item to strip comments from html files like we do in js.

**Comment by sgesin** [comment]
@vbhargavmbis `optional/translations` should be in the repo, those are label translations from core. If I remember correctly `iconSvgTemplates/buildTemplates` is included so some unit tests pass.

File: force-app/main/default/lwc/carouselImage/__tests__/carouselImage.spec.js (hunk 2 above)
Comment (sgesin) [reply force-app/main/default/lwc/carouselImage/__tests__/carouselImage.spec.js:10]: @vbhargavmbis The unit tests pass locally and lwc-recipes started including sa11y too:
https://github.com/trailheadapps/lwc-recipes/blob/master/jest-sa11y-setup.js

File: force-app/main/default/lwc/pillContainer/barePillContainer.html (hunk 3 above)
Comment (sgesin) [reply force-app/main/default/lwc/pillContainer/barePillContainer.html]: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008KfvWIAS/view

Comment (sgesin) [reply force-app/main/default/lwc/pillContainer/barePillContainer.html]: For now I'll manually strip out the comments.

File: force-app/main/default/lwc/routingService/routingService.js (hunk 4 above)
Comment (vbhargavmbis) [reply force-app/main/default/lwc/routingService/routingService.js:60]: Ah ok, I was looking at 230. Cool.

File: force-app/main/default/lwc/pillContainer/barePillContainer.html (hunk 3 above)
Comment (vbhargavmbis) [reply force-app/main/default/lwc/pillContainer/barePillContainer.html]: 👍

**Review by vbhargavmbis — APPROVED** [review]

**Pull request merged**